### PR TITLE
doc/containerized-gadgets: Update Packaging chapter to reflect current status

### PR DESCRIPTION
**Pull request by burak-ok**
# doc/containerized-gadgets: Update Packaging chapter to reflect current status

Updates the documentation about the packaging of containerized gadgets.

### CODE REVIEW

**Review by mqasimsarfraz — APPROVED** [review]
Thanks, LGTM!

File: docs/design/002-containerized-gadgets.md [hunk 1]
```diff
@@ -105,21 +105,73 @@ TODO: Should commands like `ig pull`, `ig build`, `ig push` be aliases of `ig im
...
 
-The OCI image should contain the following [OCI annotations](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys):
-- `org.opencontainers.image.title`
-- `org.opencontainers.image.description`
-- `org.opencontainers.image.url`
 
 TODO: How the user could specify these annotations when using the build command?
 
-### Definition File
+### Definition File <a name="def-file"></a>
```
Comment (blanquicet) [inline docs/design/002-containerized-gadgets.md:174]: nit: This change is not needed, you can create anchor links in md files just by doing `[Tittle of link](#name-of-section-to-link)`, in this case `[the definition file](#definition-file)`, see more [here](https://github.com/fefong/markdown_readme).

File: docs/design/002-containerized-gadgets.md [hunk 2]
```diff
@@ -105,21 +105,73 @@ TODO: Should commands like `ig pull`, `ig build`, `ig push` be aliases of `ig im
...
+}
+
+```
 
-The OCI image should contain the following [OCI annotations](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys):
-- `org.opencontainers.image.title`
-- `org.opencontainers.image.description`
-- `org.opencontainers.image.url`
 
 TODO: How the user could specify these annotations when using the build command?
```
Comment (blanquicet) [inline docs/design/002-containerized-gadgets.md:172]: I suggest to allow specifying them in the build command like [`fpm`](https://fpm.readthedocs.io/en/v1.15.1/cli-reference.html) that allows specifying `--license`, `--url`, `--description`, etc. 

However, I was wondering why not to move the author and description info into the definition file? Or maybe the convention is to have this info as annotations?

Comment (burak-ok) [reply docs/design/002-containerized-gadgets.md:172]: Nothing is set in stone and everything is still experimental - I also don't know the best place :D

I have chosen to put it in an annotation on the eBPF layer so the author is tied to the eBPF code. The definition file is "just" a file which tells Inspektor Gadget which fields of the struct should be shown -> Nothing "to author" here

Comment (blanquicet) [reply docs/design/002-containerized-gadgets.md:172]: Makes sense, so we can go with the same approach of `fpm`.

**Pull request merged**